LSPS1/README.md

```diff
@@ -0,0 +1,409 @@
+
+## Definitions
+
+### Datetime
+
+All datetimes are represented as [ISO8601](https://en.wikipedia.org/wiki/ISO_8601) in the form of `2023-02-23T08:47:30.511Z` UTC.
+
+> **Rationale** ISO8601 is human readable, it's a standard endorsed by [W3C](http://www.w3.org/TR/NOTE-datetime), and [RFC3339](https://www.rfc-editor.org/rfc/rfc3339), sortable, and supported widely.
+
+### Satoshi
+
+All satoshi values MUST be represented as a string and NOT integer values. Make sure you check your integer limits to avoid overflows.
+
+> **Rationale** Plenty of json parsers use a 32bit signed integer for integer parsing. Max safe value is 2,147,483,647; 2,147,483,647sat = BTC21.474,836,47 which is too low.
+
+### Node connection string
+
+Node connection strings like `lsp_connection_string` MUST match one of the following grammars:
+
+* `<node> '@' <ip> `:` <port>`, with the last `:` after `@` considered a separator for the port number.
+* `<node> '@' '[' <ip> ']' ':' <port>`, with any `:` inside `<ip>` considered part of the IP address.
+* `<node> '@' <tor_onion_v3_address> ':' <port>`, with the last `:` after `@` considered a separator for the port number.
+
+Word definition:
+
+* `<node>` MUST be a node id (pubkey). For example: "0200000000a3eff613189ca6c4070c89206ad658e286751eca1f29262948247a5f".
+* `<ip>` MUST be a IPv4 OR IPv6 address.
+* `<tor_onion_v3_address>` MUST be a tor onion v3 address.
+* `<port>` MUST be a [port](https://en.wikipedia.org/wiki/Port_(computer_networking)) number.
+
+### Onchain addresses
+
+- All onchain addresses MUST be a bech32 version 0 ("SegWit") or bech32m version 1 ("Taproot") address or null.
+- The lsp/user MAY support other address types.
+
+### Actors
+
+`LSP` is the API provider. `User` is the user/client of the API.
+
+### Channel sides
+
+`user_balance` are the funds on the user side. `lsp_balance` are the funds on the LSP side.
+
+### Overprovisioning
+
+The LSP is allowed to overprovision channels/onchain-payments/onchain-fees as long as it benefits the user.
+
+> **Rationale** The lsp may need to "bin" UTXOs.
+
+## Flow
+
+### API information
+
+`GET /lsp/channels` is the entrypoint for each client using the api. It lists the version of the api and all supported extensions in a dictionary.
+
+The user MUST pull `GET /lsp/channels` to read
+
+- the `version` of the api AND version of `options` and therefore prove compatibility.
+- `options` properties to determine the api boundries.
+
+Example `GET /lsp/channels` response: 
+
+```JSON
+{
+  "version": 2,
+  "website": "http://example.com/contact",
+  "options": {
+    "base_api": {
+      "version": 1,
+      "max_user_balance_satoshi": "0",
+      "max_lsp_balance_satoshi": "100000000",
+      "min_required_onchain_satoshi": null,
+      "max_channel_expiry_weeks": 24
+    }
+  }
+}
+```
+
+#### Base api options
+
+The base api itself has multiple properties that MUST be defined.
+
+```json
+"options": {
+    "base_api": {
+        "version": 1,
```

Comment:
Maybe better for this query to return a *set* of supported versions, e.g. `[1, 2]`, or for current `[1]`.  In the `POST` request the client indicates a specific version number.